src/HotChocolate/Caching/src/Caching/CacheControlConstraintsOptimizer.cs

```diff
@@ -144,6 +195,19 @@ public void OptimizeOperation(OperationOptimizerContext context)
                     constraints.Scope = directive.Scope.Value;
                     scopeSet = true;
                 }
+
+                if (directive.Vary is {Length: > 0})
+                {
+                    if (constraints.Vary != null)
+                    {
+                        constraints.Vary = constraints.Vary.Concat(directive.Vary.Select(x=>x.ToLowerInvariant())).Distinct().OrderBy(x => x).ToArray();
```

Comment:
Possibly you want to avoid Linq here and use a more efficient implementation (or more efficient collections than arrays). This is just to show the intent.
What I attempt to do here is to make the Vary headers stable w.r.t their semantics.